bundles/org.openhab.binding.paradoxalarm/src/main/java/org/openhab/binding/paradoxalarm/internal/communication/EvoCommunicator.java

```diff
@@ -49,15 +50,20 @@
 
     private MemoryMap memoryMap;
 
-    // Map of EntityTypes (Key:EntityType (Zone, Partition), Value: Map<entity IDs, Label>
     private Map<EntityType, Map<Integer, String>> entityLabelsMap = new HashMap<>();
 
     private PanelType panelType = PanelType.UNKNOWN;
-
-    public EvoCommunicator(String ipAddress, int tcpPort, String ip150Password, String pcPassword,
-            ScheduledExecutorService scheduler, PanelType panelType) throws UnknownHostException, IOException {
-        super(ipAddress, tcpPort, ip150Password, pcPassword, scheduler);
-        this.panelType = panelType;
+    private Integer maxZones;
+    private Integer maxPartitions;
+
+    private EvoCommunicator(EvoCommunicatorBuilder builder) throws UnknownHostException, IOException {

```

Comment:
I would prefer to keep the old constructor pattern (with scope `private`)

Reply:
Do you mean that in the .build() I should call the constructor with every initialization field as parameter of the constructor, taken from the builder?

Reply:
Something like this:
build() {
  EvoCOmmunicator(builder.maxZones, builder.maxPartitions, builder.ipaddress.....);
}

Reply:
Yes. But since the parameters are available in the builders, you can call without the builder-prefix